libsoftwarecontainer/lxc-softwarecontainer.in

```diff
@@ -109,50 +122,37 @@ copy_configuration()
 {
     path=$1
     rootfs=$2
-    name=$3
 
     # Set some non-static mount points, the rest are already in the config file
     echo "lxc.rootfs = $rootfs" >> $path/config
     echo "lxc.mount.entry = ${CMAKE_INSTALL_PREFIX} $rootfs/${CMAKE_INSTALL_PREFIX} none ro,bind 0 0" >> $path/config
 
-    # Bind-mount the directory containing init.lxc to /usr/sbin. This is typically
-    # /usr/sbin on the host as well.
-    INITLXCFILE=`which init.lxc`
-    if [ $? -ne 0 ]; then
-        echo "Error in template: init.lxc not found, unable to mount containing directory to /usr/sbin in container"
-        return 1
+    # Bind-mount the necessary binaries into the container
+    echo "lxc.mount.entry = $(which init.lxc) bin/init.lxc none ro,bind 0 0" >> $path/config
+    echo "lxc.mount.entry = $(which busybox) bin/busybox none ro,bind 0 0" >> $path/config
+
+    if [ ${ENABLE_NETWORKGATEWAY} = "ON" ]; then
+        echo "lxc.mount.entry = $(which iptables) bin/iptables none ro,bind 0 0" >> $path/config
     fi
-    echo "lxc.mount.entry = $(dirname $INITLXCFILE) $rootfs/usr/sbin none ro,bind 0 0" >> $path/config
 
     # If the gateway dir variable is set, add a mount entry for that one also.
-    # The gateway folder needs to be writable since wayland requires some extra
-    # files to be created by the client in the folder where the wayland socket
-    # is located
     if [ -n "$GATEWAY_DIR" ]; then
         echo "lxc.mount.entry = $GATEWAY_DIR gateways none rw,bind 0 0" >> $path/config
-        chmod go+rwx $GATEWAY_DIR
+        chmod 777 $GATEWAY_DIR
     fi
 }
 
-usage()
-{
-    echo "$1 -h|--help -b|--buffer -p|--path=<path>|--rootfs=<path>"
-    return 0
-}
-
-options=$(getopt -o hp:n: -l help,path:,name:,rootfs: -- "$@")
+options=$(getopt -o p:n: -l path:,name:,rootfs: -- "$@")
 if [ $? -ne 0 ]; then
-    usage $(basename $0)
+    echo "Usage: $(basename $0) -p|--path=<path> --rootfs=<path>"
```

Comment:
Why did we remove the usage function? Imho, it could have been expanded with more information and probably looked better than this.